Tests/RouterTests.swift

```diff
@@ -43,12 +45,25 @@ private final class RouterTestServer: NSURLProtocol {
     override func startLoading() {
         guard let requestURL = request.URL,
                   client = client else { return }
-        
-        client.URLProtocol(self, didReceiveResponse: NSHTTPURLResponse(URL: requestURL, statusCode: 200, HTTPVersion: "HTTP/1.1", headerFields: nil)!, cacheStoragePolicy: .AllowedInMemoryOnly)
-        client.URLProtocolDidFinishLoading(self)
+
+        if let canceledRequestIndex = canceledRequests.indexOf(requestURL) {
+            canceledRequests.removeAtIndex(canceledRequestIndex)
+        }
+        else {
+            client.URLProtocol(self, didReceiveResponse: NSHTTPURLResponse(URL: requestURL, statusCode: 200, HTTPVersion: "HTTP/1.1", headerFields: nil)!, cacheStoragePolicy: .AllowedInMemoryOnly)
+            client.URLProtocolDidFinishLoading(self)
+        }
     }
```

Comment:
This is a test `NSURLProtocol` just for the unit test target in order to intercept `any` outgoing calls that are not registered with Routers (some requests to `http://www.test.com` we test against). You shouldn't need to touch the `RouterTestServer` at all unless I am missing something.